lib/Runtime/PlatformAgnostic/Platform/Common/HiResTimer.cpp

```diff
@@ -56,18 +52,34 @@ namespace DateTime
             if(old_sec != new_sec)
             {
                 milliseconds = 999;
-            }  
-        }                      
+            }
+        }
 
         milliseconds = (utc_tm.tm_hour * DateTimeTicks_PerHour)
                         + (utc_tm.tm_min * DateTimeTicks_PerMinute)
                         + (utc_tm.tm_sec * DateTimeTicks_PerSecond)
                         + milliseconds;
 
-        return Js::DateUtilities::TvFromDate(1900 + utc_tm.tm_year, utc_tm.tm_mon, 
+        return Js::DateUtilities::TvFromDate(1900 + utc_tm.tm_year, utc_tm.tm_mon,
                                             utc_tm.tm_mday - 1, milliseconds);
     }
 
+    double HiResTimer::GetSystemTime()
+    {
+        ULONGLONG current = GetTickCount64();
+        ULONGLONG diff = current - data.cacheTick;
+
+        if (diff > 5) // max 5ms to respond system time changes
+        {
+            data.cacheSysTime = GetSystemTimeREAL();
```

Comment:
self-note: add `Assert(GetSystemTimeREAL() >= data.cacheSysTime + 5)` and add a slow test case